config/types_test.go

```diff
@@ -51,6 +51,9 @@ func TestDuration(t *testing.T) {
 	d = config.Duration(0)
 	require.NoError(t, d.UnmarshalTOML([]byte(`1.5`)))
 	require.Equal(t, time.Second, time.Duration(d))
+
```

Comment:
Could you add case mentioned in the original issue to make sure we error when someone inputs units with a number? e.g. `ns`